### DISCUSSION TITLE
SNES CIC

### DISCUSSION THREAD

**Discussion by mauruco** [sanni/cartreader#219] [Need help]
Hello guys.
I'm trying to dump "Super Bomberman Panic Bomber World (JP)".
As far as I know, this would be a game I need to use snesCIC.
I bought a pikit3, I can save and verify successfully, I still can't dump the game with the correct checksum.
Does anyone know what I could be doing wrong?
I didn't use the windows version, because at the moment I don't have any windows installed.
Something I noticed is that if I read the PIC12f629, its hex is a little bigger than the original, it has like a footer, I don't know if I can share it here for you to see.
![ic](https://user-images.githubusercontent.com/11234529/128921368-485c271d-0459-43e6-b3d6-09e613a843da.jpg)
![setup](https://user-images.githubusercontent.com/11234529/128921408-d701764f-8814-4d36-8b43-d6554227d885.jpg)
![pcb](https://user-images.githubusercontent.com/11234529/128921422-d32190f3-b546-44b8-83da-b5d93c1930a2.jpg)
![success](https://user-images.githubusercontent.com/11234529/128922037-c3e83418-2ba9-4962-a98e-2c91ec824e0b.png)

**Comment by mauruco** [comment]
I've already tested it on 2 SanniCartReaders I built, with 2 different ICs.

**Comment by mauruco** [comment]
I installed windows and follow the Androxilogin video: https://www.youtube.com/watch?v=7k2hb82uRVc
I was able to reproduce everything he did, write and verify was successfull.
Even so, I still can't dump successfully.
Did I forget something?
Has any functionality been removed?

**Comment by sanni** [comment]
You can read more about dumping SA-1 on this page: https://github.com/sanni/cartreader/wiki/Reading-SNES-carts

**Reply by mauruco** [reply]
My clock generator is the one from Aliexpress, listed under "Needed Parts".
I already had 1000uF capacitors here, so I used these instead of 470uF. As a power source I'm using a 5v 5a source.
These are the differences for the description "Reading SNES carts, Notes on games with SA-1".
I uploaded the sketch to calibrate the clock generator, but it only shows "read 0 correction". It doesn't matter where I plug the jumper cable.
Am I missing something?
Do I need to press a button or move a switch?

**Reply by mauruco** [reply]
PS: Measuring with the oscilloscope directly on the clock generator with the calibration sketch, CLK0 4MHz, CLK1 1MHz, CLK2 3MHz.

**Comment by sanni** [comment]
You can try an older release like V4.3 to see if that makes any difference. (V4.3 will need a FAT32 formatted SD card.)

**Reply by mauruco** [reply]
Tried v4.3 unsuccessfully, I'll leave it to try when I have other SA-1 cartridges. Even if CartReader shows the game correctly, maybe it's the game, I can't imagine anything else.
Tested with 3 different PIC12F629, 2 arduinos and v2.7, v4.3 and v6.6.

@sanni  Let me take the opportunity to get some curiosities?
1. Can I break an original cartridge if I select the wrong voltage?
2. Can I accidentally erase an original cartridge?

**Comment by sanni** [comment]
1. Maybe if you use the wrong voltage for a long time. It certainly stresses the cartridge since I have a N64 game here (Pokemon Stadium) that does not dump correctly if the voltage is wrong. 
2. Maskroms are read only and can't be erased. They can still go bad though.

**Reply by mauruco** [reply]
Thanks for everything, I love this project!

**Comment by StanMOI** [comment]
You can try the same chip on some an other board.
So you can see if problem come from your board or from your chip.

**Reply by mauruco** [reply]
I have a second one ready with a 12F629 flash on a Linux machine, which also didn't work with this cartridge.
I'm going to try to put a flashed 12F629 on a Windows machine (following the Androxilogin tutorial), with that I'll have tested 4 chips out of the 5 I bought.

**Reply by majorpbx** [reply]
Check to make sure your SNES connector slot is making good contact with your cart board edge connector. You may have to bend a few contacts to get the correct results. I had to do the same as I was chasing my tail with potentially bad traces until like I stated on another discussion that I tested from the cart board to the cart reader and found the real issue.

**Reply by mauruco** [reply]
I tested it on the second CardReader with the other 12f629, it didn't work (several times).
I took the cartridge apart and put it without a case, I tested the cartridge connections with the multimeter, it didn't work (several times).
I soldered a second capacitor parallel to the other 1000uF and it didn't work either (several times).
As I see it, the only thing I haven't tested is another SA1 cartridge, because I don't have one.
At the end of the day I'm more than 100% happy, of all my cartridges only this Japanese one didn't work, I just bought it to test it because I intend to sell the second one to get a little money back.
I still have two motherboards, someday I will test again.
Does anyone of you have this exact same cartridge and managed to dump it?

**Reply by majorpbx** [reply]
I do not, that sucks it seems to just be that cartridge. Were you able to compare your dump with a dump found online? That was suggested to me in another discussion and that's what lead to the pin/trace issue I had.

**Reply by mauruco** [reply]
I didn't get the same checksum once. I took these prints, the beginning is always good, at some point it breaks, but each attempt breaks at another point.
Three different attempts below.

![Screenshot from 2021-08-12 16-32-22](https://user-images.githubusercontent.com/11234529/129259110-b0278260-ea2e-4df6-a43d-e3366615e61c.png)
![Screenshot from 2021-08-12 16-34-07](https://user-images.githubusercontent.com/11234529/129259132-46e5b6f4-4314-4dd5-b4ae-93d9ed63c844.png)
![Screenshot from 2021-08-12 16-36-03](https://user-images.githubusercontent.com/11234529/129259148-332c120d-6b10-4ccd-b2de-349d47898002.png)
![Screenshot from 2021-08-12 16-37-09](https://user-images.githubusercontent.com/11234529/129259159-95a673d5-29c5-4bcf-9cb3-b6bd37597f4d.png)
![Screenshot from 2021-08-12 16-37-36](https://user-images.githubusercontent.com/11234529/129259174-509aa9d3-1aea-44b5-a0d6-6410f5a90c7f.png)
![Screenshot from 2021-08-12 16-39-44](https://user-images.githubusercontent.com/11234529/129259183-e088b44f-f4f5-414c-93f9-ba0f1a3b61d2.png)

**Comment by sanni** [comment]
Have you tried playing the game in a real SNES console right before dumping? Might help.

**Reply by mauruco** [reply]
I don't have the console. At the moment I only have the games.
Weekend I might be able to take this test.

**Comment by jon-nielsen** [comment]
For what it's worth, I've had similar issues with a couple of SA-1 cartridges: Not working every time on the cartreader, not working at all on a stock SNES, while ironically working on a SNES with SuperCIC. One ended up working fine as a donor cartridge - possibly because I also replaced the cap on the cart. I can also attest to the strange workaround where booting the game once on a console allowed me to dump it on the cartreader. My reason for dumping them was to verify against known checksum, so from a testing point of view I considered the cartridge as failed - not the cartreader :-)
